src/Neo.VM/EvaluationStack.cs

```diff
@@ -171,12 +171,29 @@ private static void CheckCompoundType(CompoundType rootItem, int maxItems = 2048
         {
             if (rootItem is null)
                 throw new ArgumentNullException();
+            var checker = new CompoundTypeChecker(maxItems: 2048);
+            checker.CheckCompoundType(rootItem);
+        }
+    }
 
-            var stack = new Stack<CompoundType>();
-            var visited = new HashSet<CompoundType>(ReferenceEqualityComparer.Instance);
-            int itemCount = 0;
+    internal class CompoundTypeChecker(int maxItems = 2048)
```

Comment:
One file per class please

Reply:
still struggling on it,,,,,,will clean later.